File: Cadencii/ui/java/FormAskKeySoundGenerationUiImpl.java
package org.kbinani.cadencii;

import java.awt.Dimension;
import java.awt.GridBagConstraints;
import java.awt.GridBagLayout;
import java.awt.Insets;
import javax.swing.JButton;
import javax.swing.JCheckBox;
import javax.swing.JPanel;
import org.kbinani.windows.forms.BDialog;
import org.kbinani.windows.forms.BDialogResult;
import org.kbinani.windows.forms.BLabel;

public class FormAskKeySoundGenerationUiImpl extends BDialog implements FormAskKeySoundGenerationUi
{
    private static final long serialVersionUID = 1L;
    private JPanel jPanel = null;
    private BLabel lblMessage = null;
    private JPanel jPanel1 = null;
    private JButton btnYes = null;
    private JButton btnNo = null;
    private JCheckBox chkAlwaysPerformThisCheck = null;
    private FormAskKeySoundGenerationUiListener mListener;

    public FormAskKeySoundGenerationUiImpl(
        FormAskKeySoundGenerationUiListener listener )
    {
        super();
        mListener = listener;
        initialize();
    }

    /**
     * This method initializes this
     * 
     */
    private void initialize()
    {
        this.setSize( new Dimension( 376, 190 ) );
        this.setContentPane( getJPanel() );
        setCancelButton( btnNo );
    }

    public void close(
        boolean value )
    {
        if( value ){
            super.setDialogResult( BDialogResult.CANCEL );
        }else{
            super.setDialogResult( BDialogResult.OK );
        }
    }

    /**
     * This method initializes jPanel
     * 
     * @return javax.swing.JPanel
     */
    private JPanel getJPanel()
    {
        if( jPanel == null ){
            GridBagConstraints gridBagConstraints4 = new GridBagConstraints();
            gridBagConstraints4.gridx = 0;
            gridBagConstraints4.weightx = 1.0D;
            gridBagConstraints4.fill = GridBagConstraints.HORIZONTAL;
            gridBagConstraints4.ipadx = 0;
            gridBagConstraints4.ipady = 16;
            gridBagConstraints4.insets = new Insets( 0, 0, 16, 0 );
            gridBagConstraints4.gridy = 2;
            GridBagConstraints gridBagConstraints1 = new GridBagConstraints();
            gridBagConstraints1.gridx = 0;
            gridBagConstraints1.ipady = 16;
            gridBagConstraints1.anchor = GridBagConstraints.WEST;
            gridBagConstraints1.insets = new Insets( 0, 16, 0, 0 );
            gridBagConstraints1.gridy = 1;
            GridBagConstraints gridBagConstraints = new GridBagConstraints();
            gridBagConstraints.gridx = 0;
            gridBagConstraints.weightx = 1.0D;
            gridBagConstraints.weighty = 1.0D;
            gridBagConstraints.fill = GridBagConstraints.BOTH;
            gridBagConstraints.ipadx = 0;
            gridBagConstraints.insets = new Insets( 16, 16, 8, 16 );
            gridBagConstraints.gridy = 0;
            lblMessage = new BLabel();
            lblMessage.setText( "It seems some key-board sounds are missing.\nDo you want to re-generate them now?" );
            lblMessage.setAutoEllipsis( true );
            jPanel = new JPanel();
            jPanel.setLayout( new GridBagLayout() );
            jPanel.add( lblMessage, gridBagConstraints );
            jPanel.add( getJCheckBox(), gridBagConstraints1 );
            jPanel.add( getJPanel1(), gridBagConstraints4 );
        }
        return jPanel;
    }

    /**
     * This method initializes jPanel1
     * 
     * @return javax.swing.JPanel
     */
    private JPanel getJPanel1()
    {
        if( jPanel1 == null ){
            GridBagConstraints gridBagConstraints3 = new GridBagConstraints();
            gridBagConstraints3.gridx = 0;
            gridBagConstraints3.weightx = 1.0D;
            gridBagConstraints3.gridy = 0;
            GridBagConstraints gridBagConstraints2 = new GridBagConstraints();
            gridBagConstraints2.gridx = 1;
            gridBagConstraints2.weightx = 1.0D;
            gridBagConstraints2.gridy = 0;
            jPanel1 = new JPanel();
            jPanel1.setLayout( new GridBagLayout() );
            jPanel1.add( getJButton(), gridBagConstraints2 );
            jPanel1.add( getJButton1(), gridBagConstraints3 );
        }
        return jPanel1;
    }

    /**
     * This method initializes jButton
     * 
     * @return javax.swing.JButton
     */
    private JButton getJButton()
    {
        if( btnYes == null ){
            btnYes = new JButton();
            btnYes.setText( "Yes" );
            btnYes.setPreferredSize( new Dimension( 100, 29 ) );
            btnYes.addActionListener( new java.awt.event.ActionListener()
            {
                public void actionPerformed(
                    java.awt.event.ActionEvent e )
                {
                    mListener.buttonOkClickedSlot();
                }
            } );
        }
        return btnYes;
    }

    /**
     * This method initializes jButton1
     * 
     * @return javax.swing.JButton
     */
    private JButton getJButton1()
    {
        if( btnNo == null ){
            btnNo = new JButton();
            btnNo.setText( "No" );
            btnNo.setPreferredSize( new Dimension( 100, 29 ) );
            btnNo.addActionListener( new java.awt.event.ActionListener()
            {
                public void actionPerformed(
                    java.awt.event.ActionEvent e )
                {
                    mListener.buttonCancelClickedSlot();
                }
            } );
        }
        return btnNo;
    }

    /**
     * This method initializes jCheckBox
     * 
     * @return javax.swing.JCheckBox
     */
    private JCheckBox getJCheckBox()
    {
        if( chkAlwaysPerformThisCheck == null ){
            chkAlwaysPerformThisCheck = new JCheckBox();
            chkAlwaysPerformThisCheck.setText( "Always perform this check when starting Cadencii." );
        }
        return chkAlwaysPerformThisCheck;
    }

    @Override
    public int showDialog(
        Object parent_form )
    {
        BDialogResult ret = BDialogResult.CANCEL;
        if( parent_form == null || (parent_form != null && !(parent_form instanceof BDialog)) ){
            ret = super.showDialog( null );
        }else{
            BDialog form = (BDialog)parent_form;
            ret = super.showDialog( form );
        }
        if( ret == BDialogResult.OK || ret == BDialogResult.YES ){
            return 1;
        }else{
            return 0;
        }
    }

    @Override
    public void setAlwaysPerformThisCheck(
        boolean value )
    {
        chkAlwaysPerformThisCheck.setSelected( value );
    }

    @Override
    public boolean isAlwaysPerformThisCheck()
    {
        return chkAlwaysPerformThisCheck.isSelected();
    }

    // SECTION-END-METHOD

    @Override
    public void setMessageLabelText(
        String value )
    {
        lblMessage.setText( value );
    }

    @Override
    public void setAlwaysPerformThisCheckCheckboxText(
        String value )
    {
        chkAlwaysPerformThisCheck.setText( value );
    }

    @Override
    public void setYesButtonText(
        String value )
    {
        btnYes.setText( value );
    }

    @Override
    public void setNoButtonText(
        String value )
    {
        btnNo.setText( value );
    }
} // @jve:decl-index=0:visual-constraint="10,10"

File: src/Cadencii/ui/dotnet/FormAskKeySoundGenerationUiImpl.cs
/*
 * FormAskKeySoundGenerationUiImpl.cs
 * Copyright © 2010-2011 kbinani
 *
 * This file is part of cadencii.
 *
 * cadencii is free software; you can redistribute it and/or
 * modify it under the terms of the GPLv3 License.
 *
 * cadencii is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
using System;

using System.Windows.Forms;
using cadencii.apputil;



namespace cadencii
{

    public class FormAskKeySoundGenerationUiImpl : Form, FormAskKeySoundGenerationUi
    {
        private FormAskKeySoundGenerationUiListener mListener;

        public FormAskKeySoundGenerationUiImpl(FormAskKeySoundGenerationUiListener controller)
        {
            InitializeComponent();
            mListener = controller;
            registerEventHandlers();
            Util.applyFontRecurse(this, AppManager.editorConfig.getBaseFont());
        }


        #region FormAskKeySoundGenerationUiインターフェースの実装

        /// <summary>
        /// メッセージの文字列を設定します．
        /// </summary>
        /// <param name="value">設定する文字列．</param>
        public void setMessageLabelText(string value)
        {
            lblMessage.Text = value;
        }

        public void setAlwaysPerformThisCheckCheckboxText(string value)
        {
            chkAlwaysPerformThisCheck.Text = value;
        }

        public void setYesButtonText(string value)
        {
            btnYes.Text = value;
        }

        public void setNoButtonText(string value)
        {
            btnNo.Text = value;
        }

        public void setAlwaysPerformThisCheck(bool value)
        {
            chkAlwaysPerformThisCheck.Checked = value;
        }

        public bool isAlwaysPerformThisCheck()
        {
            return chkAlwaysPerformThisCheck.Checked;
        }

        #endregion



        #region UiBaseインターフェースの実装

        public int showDialog(Object parent_form)
        {
            DialogResult ret;
            if (parent_form == null || (parent_form != null && !(parent_form is Form))) {
                ret = base.ShowDialog();
            } else {
                Form form = (Form)parent_form;
                ret = base.ShowDialog(form);
            }
            if (ret == DialogResult.OK || ret == DialogResult.Yes) {
                return 1;
            } else {
                return 0;
            }
        }

        #endregion



        #region public methods
        /// <summary>
        /// フォームを閉じます．
        /// valueがtrueのときダイアログの結果をCancelに，それ以外の場合はOKとなるようにします．
        /// </summary>
        public void close(bool value)
        {
            if (value) {
                this.DialogResult = DialogResult.Cancel;
            } else {
                this.DialogResult = DialogResult.OK;
            }
        }
        #endregion



        #region helper methods

        private static string _(string id)
        {
            return Messaging.getMessage(id);
        }

        private void registerEventHandlers()
        {
            btnYes.Click += new EventHandler(btnYes_Click);
            btnNo.Click += new EventHandler(btnNo_Click);
        }

        #endregion



        #region event handlers

        private void btnYes_Click(Object sender, EventArgs e)
        {
            mListener.buttonOkClickedSlot();
        }

        private void btnNo_Click(Object sender, EventArgs e)
        {
            mListener.buttonCancelClickedSlot();
        }
        #endregion



        #region UI implementation

        private void InitializeComponent()
        {
            this.btnNo = new Button();
            this.btnYes = new Button();
            this.chkAlwaysPerformThisCheck = new CheckBox();
            this.lblMessage = new Label();
            this.SuspendLayout();
            // 
            // btnNo
            // 
            this.btnNo.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
            this.btnNo.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.btnNo.Location = new System.Drawing.Point(183, 112);
            this.btnNo.Name = "btnNo";
            this.btnNo.Size = new System.Drawing.Size(75, 23);
            this.btnNo.TabIndex = 5;
            this.btnNo.Text = "No";
            this.btnNo.UseVisualStyleBackColor = true;
            // 
            // btnYes
            // 
            this.btnYes.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
            this.btnYes.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.btnYes.Location = new System.Drawing.Point(63, 112);
            this.btnYes.Name = "btnYes";
            this.btnYes.Size = new System.Drawing.Size(75, 23);
            this.btnYes.TabIndex = 4;
            this.btnYes.Text = "Yes";
            this.btnYes.UseVisualStyleBackColor = true;
            // 
            // chkAlwaysPerformThisCheck
            // 
            this.chkAlwaysPerformThisCheck.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Left)));
            this.chkAlwaysPerformThisCheck.AutoSize = true;
            this.chkAlwaysPerformThisCheck.Checked = true;
            this.chkAlwaysPerformThisCheck.CheckState = System.Windows.Forms.CheckState.Checked;
            this.chkAlwaysPerformThisCheck.Location = new System.Drawing.Point(14, 77);
            this.chkAlwaysPerformThisCheck.Name = "chkAlwaysPerformThisCheck";
            this.chkAlwaysPerformThisCheck.Size = new System.Drawing.Size(284, 16);
            this.chkAlwaysPerformThisCheck.TabIndex = 6;
            this.chkAlwaysPerformThisCheck.Text = "Always perform this check when starting Cadencii.";
            this.chkAlwaysPerformThisCheck.UseVisualStyleBackColor = true;
            // 
            // lblMessage
            // 
            this.lblMessage.Anchor = ((System.Windows.Forms.AnchorStyles)((((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom)
                        | System.Windows.Forms.AnchorStyles.Left)
                        | System.Windows.Forms.AnchorStyles.Right)));
            this.lblMessage.AutoEllipsis = true;
            this.lblMessage.Location = new System.Drawing.Point(12, 21);
            this.lblMessage.Name = "lblMessage";
            this.lblMessage.Size = new System.Drawing.Size(302, 53);
            this.lblMessage.TabIndex = 7;
            this.lblMessage.Text = "It seems some key-board sounds are missing. Do you want to re-generate them now?";
            // 
            // FormAskKeySoundGeneration
            // 
            this.CancelButton = this.btnNo;
            this.ClientSize = new System.Drawing.Size(326, 147);
            this.Controls.Add(this.lblMessage);
            this.Controls.Add(this.chkAlwaysPerformThisCheck);
            this.Controls.Add(this.btnNo);
            this.Controls.Add(this.btnYes);
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
            this.MaximizeBox = false;
            this.MinimizeBox = false;
            this.Name = "FormAskKeySoundGeneration";
            this.ShowIcon = false;
            this.ShowInTaskbar = false;
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.ResumeLayout(false);
            this.PerformLayout();

        }

        private Button btnNo;
        private CheckBox chkAlwaysPerformThisCheck;
        private Label lblMessage;
        private Button btnYes;

        #endregion
    }


}
